tests/providers/azure/azure_test.py

```diff
@@ -13,3 +13,410 @@
+
+    # If we specify a resource group name, the 'list_by_resource_group' method
+    # should be called.
+    FAKE_ACCOUNT.ListDisks(resource_group_name=disk.resource_group_name)
+    mock_request.assert_called_with(
+        mock.ANY,
+        'list_by_resource_group',
+        {'resource_group_name': disk.resource_group_name})
+
+  @mock.patch('libcloudforensics.providers.azure.internal.account.AZAccount.ListInstances')
+  @typing.no_type_check
+  def testGetInstance(self, mock_list_instances):
+    """Test that a particular instance from an account is retrieved."""
+    mock_list_instances.return_value = MOCK_LIST_INSTANCES
+    instance = FAKE_ACCOUNT.GetInstance('fake-vm-name')
+    self.assertEqual('fake-vm-name', instance.name)
+    self.assertEqual(
+        '/a/b/c/fake-resource-group/fake-vm-name', instance.resource_id)
+    self.assertEqual('fake-resource-group', instance.resource_group_name)
+    self.assertEqual('fake-region', instance.region)
+    self.assertEqual(['fake-zone'], instance.zones)
+
+  @mock.patch('libcloudforensics.providers.azure.internal.account.AZAccount.ListDisks')
+  @typing.no_type_check
+  def testGetDisk(self, mock_list_disks):
+    """Test that a particular disk from an account is retrieved."""
+    mock_list_disks.return_value = MOCK_LIST_DISKS
+    disk = FAKE_ACCOUNT.GetDisk('fake-disk-name')
+    self.assertEqual('fake-disk-name', disk.name)
+    self.assertEqual(
+        '/a/b/c/fake-resource-group/fake-disk-name', disk.resource_id)
+    self.assertEqual('fake-resource-group', disk.resource_group_name)
+    self.assertEqual('fake-region', disk.region)
+    self.assertEqual(['fake-zone'], disk.zones)
+
+  @mock.patch('azure.mgmt.compute.v2020_05_01.operations._disks_operations.DisksOperations.create_or_update')
+  @typing.no_type_check
+  def testCreateDiskFromSnapshot(self, mock_create_disk):
+    """Test that a disk can be created from a snapshot."""
+    mock_create_disk.return_value.done.return_value = True
+    mock_create_disk.return_value.result.return_value = MOCK_DISK_COPY
+    # CreateDiskFromSnapshot(
+    #     snapshot=FAKE_SNAPSHOT, disk_name=None, disk_name_prefix='')
+    disk_from_snapshot = FAKE_ACCOUNT.CreateDiskFromSnapshot(
+        FAKE_SNAPSHOT)
+    self.assertIsInstance(disk_from_snapshot, compute.AZDisk)
+    self.assertEqual(
+        'fake_snapshot_name_f4c186ac_copy', disk_from_snapshot.name)
+    mock_create_disk.assert_called_with(
+        FAKE_SNAPSHOT.resource_group_name,
+        'fake_snapshot_name_f4c186ac_copy',
+        mock.ANY,
+        sku='Standard_LRS')
+
+    # CreateDiskFromSnapshot(
+    #     snapshot=FAKE_SNAPSHOT,
+    #     disk_name='new-forensics-disk',
+    #     disk_name_prefix='')
+    FAKE_ACCOUNT.CreateDiskFromSnapshot(
+        FAKE_SNAPSHOT, disk_name='new-forensics-disk')
+    mock_create_disk.assert_called_with(
+        FAKE_SNAPSHOT.resource_group_name,
+        'new-forensics-disk',
+        mock.ANY,
+        sku='Standard_LRS')
```

Comment:
Are we just testing that the calls don't raise any exceptions? There doesn't seem to be any assertions after this.

Reply:
We're testing that the create disk function is called with the appropriate new disk name depending if we passed prefixes or whatnot. I did not add additional assertions on the resulting disk names as it would require us to change `mock_create_disk.return_value.result.return_value` values in-between tests which doesn't add much (the final disk name is being read from the disk created from the SDK)

Reply:
Yes, sorry I missed this.